### PR TITLE
Implement a reusable table for ephemeral data edition

### PR DESCRIPTION
**Pull request by Droniu**
## What type of PR is this?
- [ ] 💅 Refactor
- [x] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes MERX-358

## Usage Instructions, Screenshots, Recordings
This PR adds GridTable component which can be used instead of cavnas-based Datagrid in some contexts.
image

## Have you written tests?
- [x] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "@material-ui/styles": "^4.11.4",
     "@reach/auto-id": "^0.16.0",
     "@saleor/macaw-ui": "npm:@saleor/macaw-ui@0.7.4",
-    "@saleor/macaw-ui-next": "npm:@saleor/macaw-ui@1.1.3",
+    "@saleor/macaw-ui-next": "npm:@saleor/macaw-ui@0.0.0-pr-20240425162717",
```
Comment (Droniu) [inline package.json:42]: I'll replace this when https://github.com/saleor/macaw-ui/pull/793 is merged and released

File: src/components/GridTable/Body.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Box, PropsWithBox } from "@saleor/macaw-ui-next";
+import React from "react";
+
+type GridTableBodyElement = React.ElementRef<"tbody">;
+
+export const GridTableBody = React.forwardRef<GridTableBodyElement, PropsWithBox<unknown>>(
```
Comment (poulch) [inline src/components/GridTable/Body.tsx:6]: question: have you tries to type this `unknow` like this `JSX.IntrinsicElements["tbody"]`?

File: src/components/GridTable/Root.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  striped?: boolean;
+}
+
+export const GridTableRoot = React.forwardRef<GridTableRootElement, PropsWithBox<GridTableProps>>(
+  (props, forwardedRef) => {
+    const { children, striped = true, ...rest } = props;
+
+    return (
+      <GridTableContext.Provider value={{ striped }}>
+        <Box ref={forwardedRef} height={0} {...rest}>
```
Comment (poulch) [inline src/components/GridTable/Root.tsx:18]: question: why forward ref not pointing to table below?

File: src/components/GridTable/Row.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+type GridTableRowElement = React.ElementRef<"tr">;
+
+export const GridTableRow = React.forwardRef<GridTableRowElement, PropsWithBox<unknown>>(
+  ({ children, ...props }, forwardedRef) => {
+    const ctx = React.useContext(GridTableContext);
+
+    if (!ctx) {
+      throw new Error("GridTableRow must be used within a GridTableRoot");
+    }
```
Comment (poulch) [inline src/components/GridTable/Row.tsx:14]: suggestion: wrap this into hook

File: src/components/GridTable/Body.tsx (hunk 2 above)
Comment (Droniu) [reply src/components/GridTable/Body.tsx:6]: Fixed

**Review by poulch — APPROVED** [review]

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
Couple of comments:
- table components should accept only props that are inherited from Box, any other addons such as "stripped" that influence the appearance, are not needed here
- keep just the components we need, header is not needed for this table, we did couple of iterations to adjust the table and seems like there is no need for headers, we decided to not follow the exact same UI as grid has

File: src/components/GridTable/Root.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+type GridTableRootElement = React.ElementRef<"div">;
+type GridTableRootProps = GridTableProps<TableHTMLAttributes<HTMLTableElement>>;
+
+export const GridTableRoot = React.forwardRef<GridTableRootElement, GridTableRootProps>(
+  (props, forwardedRef) => {
+    const { children, ...rest } = props;
+
+    return (
+      <Box height={0} {...rest}>
```
Comment (poulch) [inline src/components/GridTable/Root.tsx:14]: suggestion: if you put all rest props here, we use the possibility to override the styles below, IMO should be spreaded on table

File: src/components/GridTable/GridTable.test.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+
+import { GridTable } from "./index";
+
+describe("GridTable", () => {
+  it("renders table with header and rows", () => {
```
Comment (andrzejewsky) [inline src/components/GridTable/GridTable.test.tsx:7]: with header?

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by Cloud11PL — APPROVED** [review]

**Pull request merged**